### PR TITLE
Project setup

### PR DESCRIPTION
**Pull request by jmarin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,84 +1,107 @@
...
+> ~re-start
+```
 
-This section should detail why people should get involved and describe key areas you are
-currently focusing on; e.g., trying to get feedback on features, fixing certain bugs, building
-important pieces, etc.
+Confirm that the platform is up and running by browsing to http://localhost:8080
+
+
+### Docker
```
Comment (hkeeler) [inline README.md]: Let's remove this until we have some Dockers.

File: README.md [hunk 2]
```diff
@@ -1,84 +1,107 @@
...
+The HMDA Platform is composed of the following modules: 
 
-Other things to include:
+### Parser
 
-  - **Technology stack**: Indicate the technological nature of the software, including primary programming language(s) and whether the software is intended as standalone or as a module in a framework or other ecosystem.
-  - **Status**:  Alpha, Beta, 1.1, etc. It's OK to write a sentence, too. The goal is to let interested people know where this project is at. This is also a good place to link to the [CHANGELOG](CHANGELOG.md).
-  - **Links to production or demo instances**
-  - Describe what sets this apart from related-projects. Linking to another doc or page is OK if this can't be expressed in a sentence or two.
+Component responsible for reading incoming data and making sure that it conforms to the HMDA File Specification
```
Comment (hkeeler) [inline README.md]: We use the term "module" in the subsequent sections.  Should we make this consistent?

File: README.md [hunk 3]
```diff
@@ -1,84 +1,107 @@
...
 
+### Data Validation
 
-**Screenshot**: If the software has visual components, place a screenshot after the description; e.g.,
+Module responsible for validating incoming data by executing validation rules as per the Edit Checks documentation
 
-![](https://raw.githubusercontent.com/cfpb/open-source-project-template/master/screenshot.png)
+### Persistence
+
+This module is responsible for persisting information into the system. It becomes the system of record for HMDA data
```
Comment (hkeeler) [inline README.md]: _~~This~~ module ~~is~~ responsible for..._ for consistency with above sections?

File: api/src/test/scala/api/protocol/HmdaApiProtocolSpec.scala [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import scala.concurrent.ExecutionContext
+import org.scalatest._
+import prop._
+import hmda.api.model.{ Status, ModelGenerators }
+import spray.json._
+
+class HmdaApiProtocolSpec extends PropSpec with PropertyChecks with MustMatchers with ModelGenerators with HmdaApiProtocol {
+
+  property("status must convert to and from json") {
+    forAll(statusGen) { (s) =>
```
Comment (hkeeler) [inline api/src/test/scala/api/protocol/HmdaApiProtocolSpec.scala]: Had to read up on [generator-driven property checks](http://www.scalatest.org/user_guide/generator_driven_property_checks) to figure out what's going on here.  I think I get it. :)  However, one thing it did mention was:

>Note that even if you are use generators that don't produce the invalid values, you still need the `whenever` clause. The reason is that once a property fails, ScalaCheck will try and shrink the values to the smallest values that still cause the property to fail. During this shrinking process ScalaCheck may pass invalid values. The `whenever` clause is still needed to guard against those values. (The `whenever` clause also clarifies to readers of the code exactly what the property is in a succinct way, without requiring that they find and understand the generator definitions.)

Do this apply here?  It's pretty much impossible for this thing to fail at this point, but maybe it's still a _best practice_?

**Comment by jmarin** [comment]
@hkeeler Thanks for reviewing this, but as you can see from my lack of assignment, the PR is not ready for merge yet. This initial PR has a different objective and will take some time to merge, I will explain at our next meeting

**Comment by hkeeler** [comment]
Cool.

**Comment by jmarin** [comment]
@hkeeler This is ready for review when you have some time

File: README.md [hunk 5]
```diff
@@ -1,84 +1,124 @@
...
-If the software includes automated tests, detail how to run those tests.
+```shell
+> projects
+[info] In file:/Users/Juan/Development/hmda-platform/
+[info] 	 * hmda
+[info] 	   model
+[info] 	   parser
 
-## Known issues
+> re-start
```
Comment (hkeeler) [inline README.md]: The `~` fell off the `~re-start` with that last commit.  Without it, we won't get the _listens for changes to underlying source code, and auto-deploys to local server_ mentioned above.

* https://github.com/spray/sbt-revolver#triggered-restart

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# Docker compose configuration file for the HMDA Platform
+# Version: 1.0.0
+
+hmda:
+  build: .
+  ports:
+    - 8080:8080
+    - 2551:2551
```
Comment (hkeeler) [inline docker-compose.yml]: What's this port for?

**Comment by hkeeler** [comment]
All reviewed up.  Just the two small fixes.

File: docker-compose.yml (hunk 6 above)
Comment (jmarin) [inline docker-compose.yml]: The default port for an ActorSystem. Will come into play shortly, although it's not being used now I wanted to include it in the configuration

**Comment by hkeeler** [comment]
Looks good.

**Pull request merged**